lib/service/Service.ts

```diff
@@ -349,6 +350,44 @@ class Service {
     return this.pool.listPeers();
   }
 
+  /**
+   * Get all trades, swaps and order status related to an order.
+   */
+  public listOrderHistory = async (args: { orderId: string }): Promise<{
+    trades: TradeInstance[], swapDeals: SwapDealInstance[], status: OrderStatus}> => {
+    const order = await this.orderBook.getOrder(args.orderId);
+
+    if (order === null) {
+      throw errors.PAIRID_NON_EXISTENT();
+    }
+
+    const trades = await this.orderBook.getTradesByOrderId(order.id, true);
+    const swapDeals = await this.swaps.getDeals(true);
+
+    const isInOrderBook = this.orderBook.getOwnOrder(order.id, order.pairId);
+    let quantity = 0;
+    let status: OrderStatus = OrderStatus.OPEN;
+    trades.forEach((trade) => {
+      quantity = quantity + trade.quantity;
+    });
+
+    if (!isInOrderBook && quantity === order.initialQuantity) {
+      status = OrderStatus.FILLED;
+    } else if (!isInOrderBook && trades.length === 0) {
+      status = OrderStatus.CANCELLED;
+    } else if (isInOrderBook && trades.length === 0) {
+      status = OrderStatus.OPEN;
+    } else if (isInOrderBook && quantity < order.initialQuantity) {
+      status = OrderStatus.PARTIALLY_FILLED;
```

Comment:
One thought that occurred to me. We may have an order that was partially filled then canceled (no longer in the order book) as well as an order that was partially filled but is still open. Should we have two different statuses for these? `PARTIALLY_FILLED` if it's open but `PARTIALLY_FILLED_CANCELED` if it's closed? Thoughts @kilrau?

Reply:
waiting @kilrau to answer

Reply:
Good point @sangaman 

> `PARTIALLY_FILLED` if it's open but `PARTIALLY_FILLED_CANCELED` if it's closed?

That actually sound good to me!

Reply:
will check it